src/Cake.Common/Tools/TextTransform/TextTransformRunner.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using Cake.Core;
+using Cake.Core.IO;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.TextTransform
+{
+    /// <summary>
+    /// The Text Transform runner.
+    /// </summary>
+    public class TextTransformRunner : Tool<TextTransformSettings>
+    {
+        private readonly ICakeEnvironment _environment;
+
+        /// <summary>
+        /// Runs Text Transform with the specified files and settings.
+        /// </summary>
+        /// <param name="sourceFile">The source file.</param>
+        /// <param name="settings">The settings.</param>
+        public void Run(FilePath sourceFile, TextTransformSettings settings)
+        {
+            if (sourceFile == null)
+            {
+                throw new ArgumentNullException("sourceFile");
+            }
+
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            Run(settings, GetArguments(sourceFile, settings));
+        }
+
+        private ProcessArgumentBuilder GetArguments(FilePath sourceFilePath, TextTransformSettings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
+
+            if (!(settings.OutputFile == null || string.IsNullOrWhiteSpace(settings.OutputFile.FullPath)))
+            {
+                builder.Append("-out");
+                builder.AppendQuoted(settings.OutputFile.MakeAbsolute(_environment).FullPath);
+            }
+
+            if (!string.IsNullOrEmpty(settings.Assembly))
+            {
+                builder.Append("-r");
+                builder.Append(settings.Assembly);
+            }
+
+            if (!string.IsNullOrEmpty(settings.Namespace))
+            {
+                builder.Append("-u");
+                builder.Append(settings.Namespace);
+            }
+
+            if (!string.IsNullOrEmpty(settings.ReferencePath))
+            {
+                builder.Append("-P");
+                builder.AppendQuoted(settings.ReferencePath);
```

Comment:
I think it would be good to make the `settings.ReferencePath` absolute before appending it.